### PR TITLE
SOC-896 | Auth pages - Don't allow redirects to external domains

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/SOC-896

* allow localUrls
* allow subdomains of ```wikia.com```
* allow subdomains of current domain

### CODE REVIEW

File: server/facets/auth/authView.ts [hunk 1]
```diff
@@ -34,7 +35,29 @@ module authView {
...
+		var redirectUrl: string = request.query.redirect || '/',
+			redirectUrlHost: string = url.parse(redirectUrl).host,
+			whiteListedDomains: Array<string> = ['wikia.com'],
+			isWhiteListedDomain: boolean;
+
+		if (!redirectUrlHost) {
+			return redirectUrl;
+		}
+
+		if (redirectUrlHost.indexOf(request.headers.host) !== -1) {
```
Comment (bkoval) [inline server/facets/auth/authView.ts]: Hey, I guess domain like "wikia.com.something.com" would pass the test. We could strenghten it to always look at the end of the host string.

File: server/facets/auth/authView.ts [hunk 2]
```diff
@@ -34,7 +35,30 @@ module authView {
...
+			redirectUrl: string = request.query.redirect || '/',
+			redirectUrlHost: string = url.parse(redirectUrl).host,
+			whiteListedDomains: Array<string> = ['wikia.com'],
+			isWhiteListedDomain: boolean;
+
+		if (!redirectUrlHost) {
+			return redirectUrl;
+		}
+
+		if (redirectUrlHost.indexOf(currentHost, redirectUrlHost.length - currentHost.length) !== -1) {
```
Comment (rafalkalinski) [inline server/facets/auth/authView.ts]: it covers situation like *wikia.com* and *glee.wikia.com*, but will give false positive result with *gleewikia.com* and *wikia.com*

File: test/specs/server/facets/auth/authView.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			host: 'glee.devbox.wikia-dev.com',
+			expected: '/',
+			description: 'Don\'t allow external domain when current domain is in URL'
+		},
+		{
+			redirect: 'http://wikia.com.google.com/',
+			host: 'wikia.com',
+			expected: '/',
+			description: 'Don\'t allow external domain when current wikia.com is put as subdomain'
+		}
```
Comment (rafalkalinski) [inline test/specs/server/facets/auth/authView.js]:
```
		{
			redirect: 'http://otherdomainwikia.com',
			host: 'wikia.com',
			expected: '/',
			description: 'Don\'t allow external domain when current wikia.com the last part of external domain'
		}
````

**Comment by bkoval** [comment]
LGTM after addressing @rafalkalinski 's comments.

**Comment by rafalkalinski** [comment]
LGTM

**Pull request merged**